torch/onnx/symbolic_opset11.py

```diff
@@ -881,6 +881,24 @@ def prim_ConstantChunk(g, self, chunks, dim):
         start = end
     return res
 
+def chunk(g, self, chunks, dim):
+    # Calculate chunk size for dynamic chunk
+    dim_size = g.op("Gather", g.op("Shape", self), dim, axis_i=0)
+    chunk_size_s = g.op("Sub", chunks, g.op("Constant", value_t=torch.tensor([1], dtype=torch.long)))
+    chunk_size = g.op("Div", g.op("Add", dim_size, chunk_size_s), chunks)
+
+    # Cases where chunk_size is divisible by number of chunks
+    chunk_vec_divisible = expand(g, chunk_size, chunks, None)
+    # Cases where chunk_size is not divisible by number of chunks
+    chunk_vec_indivisible = [expand(g, chunk_size, chunk_size_s, None),
```

Comment:
i think we can keep only the indivisible case, because when it's divisible, the values are also correct. Then it should slightly improve perf as well since these nodes are not in an if block, we will end up computing both cases anyway in the current setup.

Reply:
oh? I thought `g.op("Sub", dim_size, g.op("Mul", chunk_size, chunk_size_s))` would give 0 if it is divisble, then it should be removed, right?

Reply:
Yup, that's a great point. Removed the first branch and the where node, should be much more optimized now

Reply:
@jiafatom hmm, that's a good point to be careful with. I think it would give `chunk_size` for your case. `chunk_size_s` is guaranteed 1 smaller than the number of chunks, so 0 for last chunk shouldn't happen.